packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -261,9 +270,22 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
         // parent's clock timestamp.
         Duration nextDuration = getChallengerDuration(_challengeIndex);
 
-        // INVARIANT: A move can never be made once its clock has exceeded `GAME_DURATION / 2`
+        // INVARIANT: A move can never be made once its clock has exceeded `MAX_CLOCK_DURATION`
         //            seconds of time.
-        if (nextDuration.raw() == GAME_DURATION.raw() >> 1) revert ClockTimeExceeded();
+        if (nextDuration.raw() == MAX_CLOCK_DURATION.raw()) revert ClockTimeExceeded();
+
+        // If the remaining clock time has less than `CLOCK_EXTENSION` seconds remaining, grant the potential
+        // grandchild's clock `CLOCK_EXTENSION` seconds. This is to ensure that, even if a player has to inherit another
+        // team's clock to counter a freeloader claim, they will always have enough time to to respond. This extension
+        // is bounded by the depth of the tree. If the potential grandchild is an execution trace bisection root, the
+        // clock extension is doubled. This is to allow for extra time for the off-chain challenge agent to generate
+        // the initial instruction trace on the native FPVM.
+        if (nextDuration.raw() > MAX_CLOCK_DURATION.raw() - CLOCK_EXTENSION.raw()) {
+            // If the potential grandchild is an execution trace bisection root, double the clock extension.
+            uint64 extensionPeriod =
+                nextPositionDepth == SPLIT_DEPTH - 1 ? CLOCK_EXTENSION.raw() * 2 : CLOCK_EXTENSION.raw();
```

Comment:
Shouldn't this be `SPLIT_DEPTH + 1`?

Reply:
idts - the clock extension is for the potential grandchild. Doing this @ `SPLIT_DEPTH + 1` would imply the clock extension applies to the 3rd level in the execution trace sub-tree.

Reply:
got it. I originally thought we'd want to clock extend at the time the grandchild moves. But doing so  allows the freeloader to resolve their own subgame before the honest player has a chance to dispute it.